devito/interfaces.py

```diff
@@ -1,29 +1,102 @@
 import numpy as np
 from sympy import IndexedBase
 from tools import aligned
-
+import os
+from signal import *
+import sys
 
 __all__ = ['DenseData', 'TimeData', 'PointData']
 
 
 class DenseData(IndexedBase):
-    def __init__(self, name, shape, dtype):
+
+    def __init__(self, name, shape, dtype, disk_path=None):
         self.name = name
         self.dtype = dtype
         self.pointer = None
         self.initializer = None
+        self.disk_path = disk_path
         super(DenseData, self).__init__(name)
 
     def __new__(cls, *args, **kwargs):
         return IndexedBase.__new__(cls, args[0], shape=args[1])
 
+    # this is the default directory where memmap files will be created
+    _default_disk_path = None
+    # holds the string name of all memmap file created
+    _memmap_file_list = []
+    # default exit code used on interrupt and kill
+    _default_exit_code = 0
+
+    # functions for managing memmap files
+    # call this method to specify where you want memmap file to be created
+    @staticmethod
+    def set_default_disk_path(disk_path):
```

Comment:
When the default for a script is None, then ndarray is used instead of a memmap, but we can specify which particular object that we still want to use memmap by passing the path to __init__